src/shared/sam/cli/samCliLocalInvoke.ts

```diff
@@ -81,6 +81,10 @@ export class DefaultSamLocalInvokeCommand implements SamLocalInvokeCommand {
                     // If we have a timeout (as we do on debug) refresh the timeout as we receive text
                     params.timeout?.refresh()
                     this.logger.verbose('SAM: pid %d: stderr: %s', childProcess.pid(), removeAnsi(text))
+                    if (text.match(/Docker is not reachable/)) {
+                        vscode.window.showErrorMessage('Running AWS SAM projects locally requires Docker. Have you got it installed and running?')
+                        throw new Error('Docker is not installed or not running')
```

Comment:
`reject(new Error(...))` appears to work better on my machine.

But since this doesn't really meet the "fail fast" goal, let's revert to the `docker ps` approach.

Reply:
Reverted to the old approach.